dotnet/src/Azure.Iot.Operations.Services/AssetAndDeviceRegistry/IAdrServiceClient.cs

```diff
@@ -0,0 +1,128 @@
+﻿using Azure.Iot.Operations.Services.AssetAndDeviceRegistry.Models;
+
+namespace Azure.Iot.Operations.Services.AssetAndDeviceRegistry;
+
+/// <summary>
+/// Defines methods and events for interacting with the Asset and Device Registry (ADR) service.
+/// </summary>
+public interface IAdrServiceClient : IAsyncDisposable
+{
+    /// <summary>
+    /// Starts observing updates for a specified Asset Endpoint Profile (AEP).
+    /// </summary>
+    /// <param name="aepName">The name of the Asset Endpoint Profile to observe.</param>
+    /// <param name="cancellationToken">Token to cancel the operation.</param>
+    /// <returns>A notification response indicating the result of the operation.</returns>
+    Task<NotificationResponse> ObserveAssetEndpointProfileUpdatesAsync(string aepName, CancellationToken cancellationToken);
```

Comment:
```suggestion
    Task<NotificationResponse> ObserveAssetEndpointProfileUpdatesAsync(string aepName, CancellationToken cancellationToken = default);
```

So that users don't have to manually specify the cancellation token if they don't want it

Reply:
Also, we should have an optional param in these methods for the RPC/Telemetry layer timeout like the schema registry client has. That way, a user isn't forced to use that 10 second default you hardcoded